.github/workflows/ci.yaml

```diff
@@ -8,21 +8,44 @@ on:
     branches:
       - master
 
+defaults:
+  run:
+    shell: 'bash -o errexit -o nounset -o pipefail {0}'
+
 jobs:
   ci:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [2.5.23, 3.0.19, 4.0.15, 5.0.0-alpha-2]
     steps:
       - name: checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: jdk setup
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
+          distribution: 'temurin'
           java-version: 11
           java-package: jdk
           architecture: x64
 #      - name: unit test
-#        run: ./mvnw --batch-mode --activate-profiles indy clean test
+#        env:
+#          version: ${{ matrix.version }}
+#        run: |
+#          majorVersion=$(echo "${version}" | grep --extended-regex --only-matching "^[0-9]+")
+#          ["${majorVersion}" -gt "3"] && groupId=org.apache.groovy || groupId=org.codehaus.groovy
+#          ./mvnw --batch-mode -DgroovyVersion="${version}" clean test
 #      - name: integration test
-#        run: ./mvnw --batch-mode --activate-profiles indy -Dmaven.test.skip=true clean install invoker:install invoker:run
+#        env:
+#          version: ${{ matrix.version }}
+#        run: |
+#          majorVersion=$(echo "${version}" | grep --extended-regex --only-matching "^[0-9]+")
+#          ["${majorVersion}" -gt "3"] && groupId=org.apache.groovy || groupId=org.codehaus.groovy
+#          ./mvnw --batch-mode -DgroovyVersion="${version}" -Dmaven.test.skip=true clean install invoker:install invoker:run
       - name: full test
-        run: ./mvnw --batch-mode --activate-profiles nonindy clean install invoker:install invoker:run
+        env:
+          version: ${{ matrix.version }}
+        run: |
+          majorVersion=$(echo "${version}" | grep --extended-regex --only-matching "^[0-9]+")
+          [ "${majorVersion}" -gt "3" ] && groupId="org.apache.groovy" || groupId="org.codehaus.groovy"
+          ./mvnw --batch-mode -DgroovyVersion="${version}" -DgroovyGroupId="${groupId}" clean install invoker:install invoker:run
```

Comment:
'|' should not be used here, better use variables in github

Reply:
I'm confused how I could use variables in this case. The values are different, depending on which version of the job from the matrix is running. Ideally, I could include this along with the version in the matrix, but I didn't see a way to do that.

Reply:
I did it like that:

https://github.com/groovy/GMavenPlus/pull/268/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR46-R50

```
        run: >-
          ./mvnw ${MAVEN_ARGS}
          ${{ matrix.groovy-version != '' && format('{0}{1}', '-DgroovyVersion=', matrix.groovy-version) || '' }}
          ${{ startsWith( matrix.groovy-version , '3' ) && '-DgroovyGroupId=org.codehaus.groovy' || '' }}
          clean install invoker:install invoker:run
```

There are other ways, too. But on Github, the shell could change (e.g. ksh on AIX, zsh on Mac). So I would not want to rely on the shell.